File: SmartPhone++/SmartPhone.Services/Database/OrderItem.cs
using System;
using System.ComponentModel.DataAnnotations;

namespace SmartPhone.Services.Database
{
    public class OrderItem
    {
        [Key]
        public int Id { get; set; }
        
        [Required]
        [Range(1, int.MaxValue)]
        public int Quantity { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal UnitPrice { get; set; }
        
        [Range(0, double.MaxValue)]
        public decimal? DiscountAmount { get; set; }
        
        [Required]
        [Range(0, double.MaxValue)]
        public decimal TotalPrice { get; set; }
        
        [Required]
        [MaxLength(3)]
        public string CurrencyCode { get; set; } = string.Empty; // USD, EUR, BAM, etc.
        
        [Required]
        [MaxLength(5)]
        public string CurrencySymbol { get; set; } = string.Empty; // $, €, KM
        
        [MaxLength(200)]
        public string? ProductName { get; set; } // Snapshot of product name at time of order
        
        [MaxLength(100)]
        public string? ProductSKU { get; set; } // Snapshot of product SKU at time of order
        
        public DateTime CreatedAt { get; set; } = DateTime.UtcNow;
        
        // Foreign keys
        public int OrderId { get; set; }
        public int ProductId { get; set; }
        
        // Navigation properties
        public Order Order { get; set; } = null!;
        public Product Product { get; set; } = null!;
    }
} 

File: SmartPhone++/UI/smartphone_mobile_client/lib/model/order_item.dart
import 'package:json_annotation/json_annotation.dart';

part 'order_item.g.dart';

@JsonSerializable()
class OrderItem {
  final int id;
  final int orderId;
  final int productId;
  final int quantity;
  final double unitPrice;
  final double totalPrice;
  final String? notes;
  final String productName;
  final String productSKU;

  OrderItem({
    required this.id,
    required this.orderId,
    required this.productId,
    required this.quantity,
    required this.unitPrice,
    required this.totalPrice,
    this.notes,
    required this.productName,
    required this.productSKU,
  });

  factory OrderItem.fromJson(Map<String, dynamic> json) => _$OrderItemFromJson(json);
  Map<String, dynamic> toJson() => _$OrderItemToJson(this);

  String get formattedUnitPrice => '${unitPrice.toStringAsFixed(2)} BAM';
  String get formattedTotalPrice => '${totalPrice.toStringAsFixed(2)} BAM';
}
